db/sql/0_msar.sql

```diff
@@ -252,6 +252,32 @@ END;
 $$ LANGUAGE plpgsql RETURNS NULL ON NULL INPUT;
 
 
+CREATE OR REPLACE FUNCTION
+msar.is_default_possibly_dynamic(tab_id oid, col_id integer) RETURNS boolean AS $$/*
+Determine whether the default value for the given column is an expression or constant.
+
+If the column default is an expression, then we return 'True', since that could be dynamic. If the
+column default is a simple constant, we return 'False'. The check is not very sophisticated, and
+errs on the side of returning 'True'. We simply pull apart the pg_node_tree representation of the
+expression, and check whether the root node is a known function call type. Note that we do *not*
+search any deeper in the tree than the root node. This means we won't notice that some expressions
+are actually constant (or at least static), if they have a function call or operator as their root
+node.
```

Comment:
I did quite a bit of research about this. It's not possible to actually access a navigable tree form of the `pg_node_tree` type in SQL (using PG core functionality). All the functions for doing that are C functions which are inaccessible from the DB.

The only way to improve, therefore, would involve learning enough about this poorly-documented internal format to actually construct a tree, and the advantage would be that we could notice (for example) if a pure function had only constant inputs, and call that output therefore constant.

It might make an interesting research project someday...